com.civicrm.hrjobroles/CRM/Hrjobroles/Import/Parser/HrJobRoles.php

```diff
@@ -418,26 +418,20 @@ function fini() {}
 
 
   /**
-   * get the option database ID given its label or ID
+   * get the Option Value database value given its label
    * @param String|Integer $option
-   * @param String|Integer $value
-   * @return Integer
+   * @param String $value
+   * @return String|NULL
    * @access private
    */
-   private function getOptionKey($option, $value)  {
-     if (is_numeric ($value)) {
-       $search_field = 'id';
-     }
-     else {
-       $search_field = 'label';
-     }
-     $index = array_search(strtolower($value), array_column($this->_optionsList[$option], $search_field));
-     if ($index !== FALSE)  {
-       return (int) $this->_optionsList[$option][$index]['id'];
-     }
-     else {
-       return 0;
-     }
+  private function getOptionKey($option, $value)  {
+    $index = array_search(strtolower($value), array_column($this->_optionsList[$option], 'label'));
+    if ($index !== FALSE)  {
+      return $this->_optionsList[$option][$index]['value'];
+    }
+    else {
+      return NULL;
+    }
```

Comment:
The output format of buildOptions can make this piece of code a little bit simpler. 
Civi also has the CRM_Utils_Array::key method that can be used to avoid the if/else. It would be something like this:

```php
private function getOptionKey($option, $value) {
    return CRM_Utils_Array::key($value, $this->_optionsList[$option]);
}
```